pandas/core/generic.py

```diff
@@ -4988,7 +4998,7 @@ def sample(
             )
 
         locs = rs.choice(axis_length, size=n, replace=replace, p=weights)
-        return self.take(locs, axis=axis, is_copy=False)
+        return self.take(locs, axis=axis)
```

Comment:
I think you could take a look here: the default is `None`, and if `None`, `is_copy` will be set to `True`. Here `is_copy` was originally `False`, but as you removed `is_copy`, this will be set to `True`.

Reply:
Tried reverting that change and it still fails even if `is_copy=False` .... seems the initial change of default `is_copy` from `True` to   `None` is breaking those tests  -- prob something silly ive done  with the if else condition

Reply:
right, i meant if those are intended, you might need to take care of warning message changes caused by it

Reply:
ah ok thanks - yeh think i found a workaround now.